things/service.go

```diff
@@ -439,3 +477,38 @@ func (ts *thingsService) members(ctx context.Context, token, groupID, groupType
 	}
 	return res.Members, nil
 }
+
+func (ts *thingsService) authorize(ctx context.Context, subject, object string, relations ...string) error {
+	req := &mainflux.AuthorizeReq{
+		Sub: subject,
+		Obj: object,
+	}
+	for _, relation := range relations {
+		req.Act = relation
+		res, err := ts.auth.Authorize(ctx, req)
+		if err != nil {
+			return errors.Wrap(ErrAuthorization, err)
+		}
+		if res.Authorized {
+			return nil
+		}
+	}
+	return ErrAuthorization
+}
+
+func (ts *thingsService) addPolicy(ctx context.Context, subject, object, relation string) error {
```

Comment:
Why both Things service and Auth service have `AddPolicy()`? Should not we add policies only on one place?

Reply:
Yes, AddPolicy should belong to Auth service. However, I wrote this `addPolicy` to prevent code-duplication while calling `auth.AddPolicy`. So, `addPolicy` just calls the auth's AddPolicy. Although `addPolicy` here is only used once, we will need to call it more than once for further policies (e.g. creating a channel). Should it be separated from `ts *thingsService`?

Reply:
If this is just calling Auth service, than a gRPC client for Auth service should be made and we can call directly Auth service with it.